snewpdag/plugins/MeanDist.py

```diff
@@ -11,52 +10,70 @@
     detector: string, "detector name, ordering" ,
               one of ["IceCube, NO","IceCube, IO","HK, NO","HK, IO","SK, NO","SK, IO",
               "DUNE, NO","DUNE, IO","JUNO, NO","JUNO, IO"]
-    in_field: string, "count",
+    in_field: string, "n" or "n_norm",
               to get the count numbers from data["count"]
-    out_field: string, "dist" (as an example),
+    out_field: string, "dist" or "dist_norm",
               used for adding/updating the field in the data dict
-    t0:       the "measured/estimated" t_nought
-              
+    t0:       the "measured/estimated" time of the start of SN emission (ms)
+    mode:     string, either "Histogram" or "Error",
+              "Histogram" if output is used to plot histogram for a fixed true distance,
+              "Error" if output is used to plot errors against true distance
+
 '''
 
 import logging
 import numpy as np
 from snewpdag.dag import Node
 from snewpdag.plugins import DistCalc1
 from snewpdag.plugins import DistCalc2
+from snewpdag.plugins import DistNormalizer
 
 class MeanDist(Node):
     
-    def __init__(self, detector, in_field, out_field, t0, **kwargs):
+    def __init__(self, detector, in_field, out_field, t0, mode, **kwargs):
         self.detector = detector
         self.in_field = in_field
         self.out_field = out_field
         self.t0 = t0
+        self.mode = mode
         super().__init__(**kwargs)
     
     def mean_dist(self, data):
         h1 = DistCalc1(self.detector, self.in_field, self.out_field, self.t0, name = self.name)
-        h2 = DistCalc2(self.detector, self.in_field, self.out_field, self.t0, name = self.name)  
+        h2 = DistCalc2(self.detector, self.in_field, self.out_field, self.t0, name = self.name) 
+        
+        if self.mode == "Histogram":
+            dist1, dist1_err = h1.dist_calc1(data)
+            dist2, dist2_err = h2.dist_calc2(data)
 
-        dist1, dist1_err = h1.dist_calc1(data)
-        dist2, dist2_err = h2.dist_calc2(data)
+            mdist = (dist1/dist1_err**2 + dist2/dist2_err**2)/ (1.0/dist1_err**2+1.0/dist2_err**2)
+            mdist_err = 1.0/np.sqrt(1.0/dist1_err**2+1.0/dist2_err**2)
+    
+            return mdist, mdist_err
         
-        mdist = (dist1/dist1_err**2 + dist2/dist2_err**2)/ (1.0/dist1_err**2+1.0/dist2_err**2)
-        mdist_err = 1.0/np.sqrt(1.0/dist1_err**2+1.0/dist2_err**2)
+        elif self.mode == "Error":
+            true_dist_array = np.linspace(4,40,100,endpoint=True)
+            mdist_array = np.array([])
+            mdist_err_array = np.array([])
+            for i in range(100):
+                h3 = DistNormalizer(true_dist_array[i], self.in_field, self.out_field, name = self.name)
```

Comment:
Please note that when you run a dag, you will probably run into a TypeError saying this line is calling a module instead of a class. I've pushed it anyway so that perhaps someone could help resolve it.